app/assets/javascripts/views/account_switcher/account_switcher.ts

```diff
@@ -0,0 +1,105 @@
+import { ApplicationGroup } from '@/ui_models/application_group';
+import { WebApplication } from '@/ui_models/application';
+import template from './account-switcher.pug';
+import {
+  ApplicationDescriptor,
+} from 'snjs';
+import { PureViewCtrl } from '@Views/abstract/pure_view_ctrl';
+import { WebDirective } from '@/types';
+
+class AccountSwitcherCtrl extends PureViewCtrl<{}, {
+  descriptors: ApplicationDescriptor[];
+  editingDescriptor?: ApplicationDescriptor
+}> {
+  private $element: JQLite
+  application!: WebApplication
+  private removeAppGroupObserver: any;
+  /** @template */
+  activeApplication!: WebApplication
+
+  /* @ngInject */
+  constructor(
+    $element: JQLite,
+    $timeout: ng.ITimeoutService,
+    private mainApplicationGroup: ApplicationGroup
+  ) {
+    super($timeout);
+    this.$element = $element;
+    this.removeAppGroupObserver = mainApplicationGroup.addApplicationChangeObserver(() => {
+      this.activeApplication = mainApplicationGroup.primaryApplication as WebApplication;
+      this.reloadApplications();
+    });
```

Comment:
Since deinit is called during $onDestroy this should probably be inside $onInit instead